temporal-sdk/src/main/java/io/temporal/common/converter/SearchAttributesUtil.java

```diff
@@ -0,0 +1,187 @@
+          log.error("Error parsing Search Attribute: {} of type {}.", key, type);
+        } else {
+          deserializedMap.put(
+              key, converter.fromData(attribute.getValue(), javaType.getClass(), javaType));
+        }
+      }
+    }
+    return deserializedMap;
+  }
+
+  public static boolean parseDefaultSearchAttributes(
+      String key, Payload value, Map<String, Object> deserializedMap) {
+    DefaultSearchAttributes searchAttribute;
+    try {
+      searchAttribute = DefaultSearchAttributes.valueOf(key);
+    } catch (IllegalArgumentException e) {
+      return false;
+    }
+    switch (searchAttribute) {
+      case BatcherNamespace:
+      case BatcherUser:
+      case BinaryChecksums:
+      case ExecutionStatus:
+      case RunId:
+      case TaskQueue:
+      case TemporalChangeVersion:
+      case WorkflowId:
+      case WorkflowType:
+        deserializedMap.put(key, converter.fromData(value, String.class, String.class));
+        break;
+      case CloseTime:
+      case ExecutionTime:
+      case StartTime:
+        deserializedMap.put(
+            key, converter.fromData(value, LocalDateTime.class, LocalDateTime.class));
+        break;
+      case ExecutionDuration:
+      case HistoryLength:
+      case StateTransitionCount:
+        deserializedMap.put(key, converter.fromData(value, Integer.class, Integer.class));
+        break;
+    }
+    return true;
+  }
+
+  public static Type encodedTypeToJavaType(String type) {
+    SearchAttributeType attributeType;
+    try {
+      attributeType = SearchAttributeType.valueOf(type);
+    } catch (IllegalArgumentException e) {
+      return null;
+    }
+    switch (attributeType) {
+      case String:
+      case Keyword:
+        return String.class;
+      case Int:
+        return Integer.class;
+      case Double:
+        return Double.class;
+      case Bool:
+        return Boolean.class;
+      case Datetime:
+        return LocalDateTime.class;
+    }
+    return null;
+  }
+
+  public static SearchAttributeType javaTypeToEncodedType(Class type) {
+    if (String.class.equals(type)) {
+      return SearchAttributeType.String;
+    } else if (Integer.class.equals(type)) {
+      return SearchAttributeType.Int;
+    } else if (Double.class.equals(type) || Float.class.equals(type)) {
+      return SearchAttributeType.Double;
+    } else if (Boolean.class.equals(type)) {
+      return SearchAttributeType.Bool;
+    } else if (LocalDateTime.class.equals(type)) {
+      return SearchAttributeType.Datetime;
+    }
+    return SearchAttributeType.Unspecified;
+  }
+
+  enum DefaultSearchAttributes {
+    BatcherNamespace,
+    BatcherUser,
+    BinaryChecksums,
+    CloseTime,
+    ExecutionDuration,
+    ExecutionStatus,
+    ExecutionTime,
+    HistoryLength,
+    RunId,
+    StartTime,
+    StateTransitionCount,
+    TaskQueue,
+    TemporalChangeVersion,
+    WorkflowId,
+    WorkflowType
+  }
+
+  enum SearchAttributeType {
+    Unspecified,
```

Comment:
This class should be a part of DataConverter, I don't see a good reason to leak this detail our from the DataConverter.

Reply:
This comment stays valid